tests/test_streams.py

```diff
@@ -0,0 +1,256 @@
+
+    with open("tests/output_streams/write_varint.out", "rb") as stream:
+        assert stream.read() == encode_varint(over32)
+
+    # Test 64-bit unsigned integer limit write capabilities
+    with open("tests/output_streams/write_varint.out", "wb") as stream:
+        write_varint(stream, maximum)
+        stream.flush()
+
+    with open("tests/output_streams/write_varint.out", "rb") as stream:
+        assert stream.read() == encode_varint(maximum)
+
+
+def test_write_varint_stream_non_binary():
+    with open("tests/output_streams/non_binary.out", "w") as stream, \
+            pytest.raises(TypeError):
+        write_varint(stream, 1)
+
+
+def test_write_varint_stream_non_writable():
+    with open("tests/output_streams/non_binary.out", "rb") as stream, \
+            pytest.raises(TypeError):
+        write_varint(stream, 1)
+
+
+def test_write_varint_multiple(output_dir):
+    with open("tests/output_streams/write_varint_multiple.out", "wb") as stream:
+        # Write the same contents as varints.in to a file
+        write_varint(stream, 1)
+        write_varint(stream, 123456789)
+        write_varint(stream, 1)
+        write_varint(stream, 3000000000)
+        write_varint(stream, 18446744073709551615)
+
+        stream.flush()
+
+    # Check that both files have the same contents, as expected
+    with open("tests/output_streams/write_varint_multiple.out", "rb") as testing, open(
+            "tests/streams/varints.in", "rb"
+    ) as expected:
+        assert testing.read() == expected.read()
+
+
+def test_message_write_to(output_dir):
+    with open("tests/output_streams/write_message.out", "wb") as stream:
+        # Write test message to file
+        stream.write(b"\x10")  # Message size prefix, same as given in multiple.in
+        oneof_example.write_to(stream)
+
+        stream.flush()
+
+    # Check written message against known-good wire of the same message
+    with open("tests/output_streams/write_message.out", "rb") as testing, open(
+            "tests/streams/multiple.in", "rb"
+    ) as expected:
+        assert testing.read(17) == expected.read(17)
+
+
+def test_message_write_to_stream_non_binary():
+    with open("tests/output_streams/non_binary.out", "w") as stream, \
+            pytest.raises(TypeError):
+        oneof_example.write_to(stream)
+
+
+def test_message_write_to_stream_non_writable():
+    with open("tests/output_streams/non_binary.out", "rb") as stream, \
+            pytest.raises(TypeError):
+        oneof_example.write_to(stream)
+
+
+def test_message_write_to_multiple(output_dir):
+    with open("tests/output_streams/write_message_multiple.out", "wb") as stream:
+        # Write test messages to file separately
+        stream.write(b"\x10")  # Message size prefix, same as given in multiple.in
+        oneof_example.write_to(stream)
+        stream.write(b"\x0E")  # Message size prefix, same as given in multiple.in
+        nested_example.write_to(stream)
+
+        stream.flush()
```

Comment:
`.close` flushes the stream - see https://docs.python.org/3/library/io.html#io.IOBase.close
It looks like we can remove all `.flush` invocations that occur at the end of `with` blocks

Reply:
Fixed